components/dashboard/src/user-settings/SelectIDE.tsx

```diff
@@ -33,27 +32,44 @@ export default function SelectIDE(props: SelectIDEProps) {
         user?.additionalData?.ideSettings?.useLatestVersion ?? false,
     );
 
-    const actualUpdateUserIDEInfo = async (selectedIde: string, useLatestVersion: boolean) => {
-        const additionalData = user?.additionalData ?? {};
-        const settings = additionalData.ideSettings ?? {};
-        settings.settingVersion = "2.0";
-        settings.defaultIde = selectedIde;
-        settings.useLatestVersion = useLatestVersion;
-        additionalData.ideSettings = settings;
+    const actualUpdateUserIDEInfo = useCallback(
+        async (selectedIde: string, useLatestVersion: boolean) => {
+            const additionalData = user?.additionalData || {};
+            const ideSettings = additionalData.ideSettings || {};
 
-        const newUserData = await updateUser.mutateAsync({ additionalData });
-        props.updateUserContext && setUser({ ...newUserData });
-    };
+            // Avoid mutating user object in state for updates
```

Comment:
This comment is a bit confusing without seeing the previous code. Maybe just remove it?